components/nexus-core/src/main/java/org/sonatype/nexus/internal/app/BaseUrlManagerImpl.java

```diff
@@ -45,11 +45,9 @@ public BaseUrlManagerImpl(final Provider<HttpServletRequest> requestProvider) {
     this.requestProvider = checkNotNull(requestProvider);
   }
 
-  // FIXME: Sort out persistence and synchronization
+  private volatile String url;
 
-  private String url;
-
-  private boolean force;
+  private volatile boolean force;
```

Comment:
mostly just sanity since the data is otherwise not protected by synchronization.  may be uneeded, but unsure it hurts